### ISSUE TITLE
private members of objects can be accessed from console

### ISSUE THREAD

**Issue by Odamra** [Microsoft/TypeScript#27290]
**TypeScript Version:**  2.7.2

**Search Terms:**
expose, export, window, object, security 
**Code**
lets say that you have a class with a private member 

![image](https://user-images.githubusercontent.com/43512514/45925719-a652b980-bf23-11e8-80d2-71f4c8ef6670.png)

 if we expose an object of this class

![image](https://user-images.githubusercontent.com/43512514/45925731-dac67580-bf23-11e8-8609-66a5797ce03b.png)

any user can use the private member of the object in the console
![image](https://user-images.githubusercontent.com/43512514/45925767-87a0f280-bf24-11e8-9bf7-4d9658b47f2c.png)

**Expected behavior:**
private members should not be accessible 

**Actual behavior:**
any user can use the private member of the object in the console

**Comment by Hoishin** [comment]
As far as I know, there is not such thing as private property in JavaScript (yet) and the `private` only means that it is private in TypeScript world (before compilation).

**Comment by Odamra** [comment]
> As far as I know, there is not such thing as private property in JavaScript (yet) and the `private` only means that it is private in TypeScript world (before compilation).

but that means every private members in a class of typescript will go public after compilation
that's sounds security issue
especially for the projects which expose interfaces for its users

**Comment by Hoishin** [comment]
Yes, that's why the private property proposal is worked on for ECMAScript. 
https://github.com/tc39/proposal-private-methods
https://github.com/tc39/proposal-class-fields
https://github.com/tc39/proposal-static-class-features/

I don't think TypeScript can do anything about this.
Maybe TS can do something like putting random unique hash to each property or something? But that does not hide the property, just a different name.

**Comment by Odamra** [comment]
> Yes, that's why the private property proposal is worked on for ECMAScript.
> https://github.com/tc39/proposal-private-methods
> https://github.com/tc39/proposal-class-fields
> https://github.com/tc39/proposal-static-class-features/
> 
> I don't think TypeScript can do anything about this.
> Maybe TS can do something like putting random unique hash to each property or something? But that does not hide the property, just a different name.

I see, Thanks for explaining my friend
